test/EndToEndTests/Tests/Client/AsynchronousTests/AsyncMethodTests.cs

```diff
@@ -135,6 +136,51 @@ public async Task SaveChangesTest()
             this.EnqueueTestComplete();
         }
 
+        [Fact, Asynchronous]
+        public async Task JsonBatchSequencingTest()
+        {
+            DefaultContainer context = this.CreateWrappedContext<DefaultContainer>().Context;
+            Customer c1 = new Customer { CustomerId = 1, Name = "customerOne" };
+            Customer c2 = new Customer { CustomerId = 2, Name = "customerTwo" };
+            context.AddToCustomer(c1);
+            context.AddToCustomer(c2);
+            await context.SaveChangesAsync(SaveChangesOptions.BatchWithSingleChangeset);
+
+            c1.Name = "customerOne updated name";
+            c2.Name = "customerTwo updated name";
+
+            context.UpdateObject(c1);
+            context.UpdateObject(c2, c1.CustomerId);
+
+            var dscResponse = await context.SaveChangesAsync(SaveChangesOptions.BatchWithIndependentOperations | SaveChangesOptions.UseJsonBatch);
+            Assert.Equal(204, (dscResponse.First() as ChangeOperationResponse).StatusCode);

```

Comment:
[Question] Shouldn't we be checking the second item as well as it had the depends on property?

Reply:
Good catch!

Reply:
Resolved